docs/docs/guide/8-shards-architecture.md

```diff
@@ -0,0 +1,552 @@
+# Shards Architecture
```

Comment:
Ok so for efficient injection by llms we really need short pages and many of them vs big pages.
Split by sections basically please, an alternative would be that we post process them automatically but I think it's just easier to do it as we document.